src/main/java/com/greenfox/poker/service/UserService.java

```diff
@@ -2,77 +2,114 @@
+    return new StatusError("fail", "email address already exists");
   }
 
-  public ResponseEntity<?> respondToInvalidUsernameOrPassword(String errorMessage) {
-    StatusError statusError = new StatusError("fail", errorMessage);
-    return new ResponseEntity(statusError, HttpStatus.UNAUTHORIZED);
+  public ResponseType registerWithOccupiedUsername() {
+    return new StatusError("fail", "username already exists");
   }
 
-  public ResponseEntity<?> register(PokerUser pokerUser,
-          BindingResult bindingResult) {
-    if (bindingResult.hasErrors()) {
-      return respondToMissingParameters(bindingResult);
-    } else if (isEmailOccupied(pokerUser)) {
-      return respondToOccupiedConflict("email address already exists");
-    } else if (isUsernameOccupied(pokerUser)) {
-      return respondToOccupiedConflict("username already exists");
+  public boolean isLoginValid(LoginRequest loginRequest){
+    List<PokerUser> users = pokerUserRepo.findByUsername(loginRequest.getUsername());
+    if (users.size() > 0 && users.get(0).getPassword().equals(loginRequest.getPassword())){
+      return true;
     }
-    return mockRespondToSuccessfulRegisterOrLogin();
+    return false;
   }
 
-  public ResponseEntity<?> login(BindingResult bindingResult,
-          LoginRequest loginRequest) {
-    if (bindingResult.hasErrors()) {
-      return respondToMissingParameters(bindingResult);
-    } else if ((!loginRequest.getUsername().equals("Bond") || !loginRequest
-            .getPassword().equals("password123"))) {
-      return respondToInvalidUsernameOrPassword("invalid username or password");
-    }
-    return mockRespondToSuccessfulRegisterOrLogin();
+  public ResponseType loginWithIvalidUsernameOrPassword() {
+    return new StatusError("fail", "invalid username or password");
   }
 
   public boolean isEmailOccupied(PokerUser pokerUser) {
-    if (pokerUser.getEmail().equals("occupied@email.com")) {
-      return true;
-    }
-    return false;
+    boolean isEmailOccupied = false;
+    isEmailOccupied = pokerUserRepo.existsByEmail(pokerUser.getEmail());
+    return isEmailOccupied;
   }
 
   public boolean isUsernameOccupied(PokerUser pokerUser) {
-    if (pokerUser.getUsername().equals("occupiedUserName")) {
+    boolean isUsernameOccupied = false;
+    isUsernameOccupied = pokerUserRepo.existsByUsername(pokerUser.getUsername());
+    return isUsernameOccupied;
+  }
+
+  public long getUserIdFromUsername(String username){
+    return pokerUserRepo.findByUsername(username).get(0).getId();
+  }
+
+  public boolean isUserExistsInDB(long id) {
+    if (pokerUserRepo.exists(id)) {
       return true;
+    } else {
+      return false;
     }
-    return false;
+  }
+
+  public PokerUserDTO getUserDTO(long id) {
+    PokerUser pokerUser = pokerUserRepo.findOne(id);
+    pokerUserDTO.setId(id);
+    pokerUserDTO.setUsername(pokerUser.getUsername());
+    pokerUserDTO.setAvatar(pokerUser.getAvatar());
+    pokerUserDTO.setChips(pokerUser.getChips());
```

Comment:
You should have a method for this purpose in your service layer which converts a PokerUser object to a PokerUserDTO object.